dora/core/server/worker/src/main/java/alluxio/worker/netty/PacketReadTaskStateMachine.java

```diff
@@ -248,9 +248,10 @@ private void onReadingData() {
     // too many  in-flight packets, and we need to wait some time.
     LOG.debug("Putting an object into the flow control queue.");
     SendDataEvent sendDataEvent = new SendDataEvent(start, packetSize, packet);
-    mFlowControlQueue.offer(sendDataEvent);
-    LOG.debug("An object has been put into the flow control queue successfully.");
-
+    if (packet != null) {
```

Comment:
when would packet be null? And we created a SendDataEvent already?

Reply:
packet is null only if `getDataBuffer` return null. If packet is null, there will be an infinite loop, because at this time a `SendDataEvent` is put into the blocking queue, but it will never come out. It doesn't matter that `SendDataEvent` is created because the `onSendingData` method will skip it if it find that the packet in `SendDataEvent` is null.